### DISCUSSION TITLE
Neorg command not available after installing

### DISCUSSION THREAD

**Discussion by JeroenDedimo** [nvim-neorg/neorg#1026] [Q&A] [answered]
Hi,

I'm probably doing it wrong. But I have trouble finding a conscise installation instruction with lazy.nvim that just works.

My steps so far:

- Install neovim on Ubuntu linux
- Install NvChad  
  
-> Everything works perfectly at this point. I'm a happy camper. Just wish I could try out Neorg ;)  
  
- Create new file ~/.config/nvim/lua/plugins/neorg.lua
- Paste the code below in the new file
- Restart neovim
- Try the Neorg command: not found

```
require("lazy").setup({
  {
    "nvim-neorg/neorg",
    build = ":Neorg sync-parsers",
    dependencies = { "nvim-lua/plenary.nvim" },
    config = function()
      require("neorg").setup {
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes",
              },
            },
          },
        },
      }
    end,
  },
})
```

Any help would be greatly appreciated. Neorg seems like a really cool project!

**Comment by champignoom** [comment]
Try `:Lazy`. Is `neorg` in the `Loaded` part?

**Reply by JeroenDedimo** [reply]
Neorg is not in Loaded part, nor in the Not_Loaded part for that matter. 

I'm going to try max397574's advice and then check again.

**Comment by max397574** [comment]
I'm quite sure you should just do  `return { "nvim-neorg/neorg", ... }` in that file
no clue how exactly nvchad organizes this though

**Reply by JeroenDedimo** [reply]
Thanks Max! I suspicious of the require.... from the start, but didn't know how to hack up or return the data part (Just starting to learn Lua).

**Comment by JeroenDedimo** [comment]
Thanks so much for the kind answers to my absolutely n00b question. I realize I have a long way to go to understand neovim config! 

I'll try out the above right now.

**Comment by JeroenDedimo** [comment]
I don't know if this is relevant. But it does worry me. The output of :checkhealth lazy is:

```
- OK Git installed
- OK no existing packages found by other package managers
- OK packer_compiled.lua not found
```
Especially the last line about packer. Can this be the cause?

After replacing my plugin loading code with this: I stil don't see Neorg in :Lazy out put.

```
return {
    "nvim-neorg/neorg",
    build = ":Neorg sync-parsers",
    dependencies = { "nvim-lua/plenary.nvim" },
    config = function()
      require("neorg").setup {
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes",
              },
            },
          },
        },
      }
    end,
  }
```

**Comment by benatouba** [comment]
The line about packer_compiled not found is correct and good for you ;) 
[packer.nvim](https://github.com/wbthomason/packer.nvim) is a different package manager that uses these files.

NvChad uses [lazy.nvim](https://github.com/folke/lazy.nvim) which is the package manager of choice for most (new) users now. Packer is/was great too though.

**Comment by benatouba** [comment, marked as answer]
Try the following:

```lua
local plugins = {
{
    "nvim-neorg/neorg",
    build = ":Neorg sync-parsers",
    dependencies = { "nvim-lua/plenary.nvim" },
   lazy = false, -- if you want to load on startup and eagerly
    config = function()
      require("neorg").setup {
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes",
              },
            },
          },
        },
      }
    end,
  }
}
return plugins
```

You return a table of tables of plugin configs.

This should give you the right structure and enables you to add more custom plugins later.

Also your file path should be `~/.config/nvim/lua/custom/plugins.lua` (At least that is what I get from the docs).

Edit: you might also need a file `custom/chadrc.lua` with the contents
```lua
M.plugins = "custom.plugins"
```
the [docs](https://nvchad.com/docs/config/plugins) are not 100% clear about that.

**Reply by JeroenDedimo** [reply]
I cannot thank you enough! This worked perfectly!

The main error in my approach was: wanting to get functionality without understanding config first.

The second concrete error was: installing NvChad, then ignoring it's docs and going straight to lazy.nvim to try and understand plugins. 

You're absolutely awesome for taking the time to answer a n00b question like this!

**Comment by benatouba** [comment]
This is one of the issues of using a pre-configured distribution. Although it is great to have a lot of functionality from the start, they stick you into a more rigid system.

Vims/Neovims strength lies in personalization. Set up every part of your editor to suit your needs. [PDE](https://www.youtube.com/watch?v=QMVIJhC9Veg) 

I suggest setting up everything the way you like it, once you know your way around.

**Reply by JeroenDedimo** [reply]
Haha completely agree! This made me realize what I did to myself starting with Chad. 

I'll try your solution above this weekend and in a few weeks schedule a deep dive in bare neovim / config / lua to make sure I know all the nooks and crannies and retain my maneuverability in neovim.

**Reply by benatouba** [reply]
[kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim)

Many say that this is a good starting point. :)
